harness/harness.rb

```diff
@@ -1,51 +1,39 @@
-require 'csv'
+require 'benchmark'
 
 # Warmup iterations
-$WARMUP_ITRS = ENV.fetch('WARMUP_ITRS', 15).to_i
+WARMUP_ITRS = ENV.fetch('WARMUP_ITRS', 15).to_i
 
 # Minimum number of benchmarking iterations
-$MIN_BENCH_ITRS = 10
+MIN_BENCH_ITRS = ENV.fetch('MIN_BENCH_ITRS', 10).to_i
 
 # Minimum benchmarking time in seconds
-$MIN_BENCH_TIME = 10
+MIN_BENCH_TIME = ENV.fetch('MIN_BENCH_TIME', 10).to_i
 
-$out_csv_path = File.expand_path(ENV.fetch('OUT_CSV_PATH', 'output.csv'))
+default_path = "results-#{RUBY_ENGINE}-#{RUBY_ENGINE_VERSION}-#{Time.now.strftime('%F-%H%M%S')}.csv"
```

Comment:
The problem with this is that it will create a large number of output files, one for every run. IMO specifying the output path as an input was the right way to go.